modules/javafx.controls/src/test/java/test/javafx/scene/control/ListViewTest.java

```diff
@@ -2183,4 +2184,246 @@ private void attemptGC(WeakReference<? extends Object> weakRef, int n) {
+
+            }
+        } else {
+            listView.getItems().addAll(heights);
+            Toolkit.getToolkit().firePulse();
+
+        }
+        listView.scrollTo(scrollToIndex);
+        Toolkit.getToolkit().firePulse();
+        if (layoutTwice) {
+            listView.requestLayout();
+            Toolkit.getToolkit().firePulse();
+        }
+        if (selectIndex) {
+            listView.getSelectionModel().select(scrollToIndex);
+            Toolkit.getToolkit().firePulse();
+        }
+
+        verifyListViewScrollTo(listView, listViewHeight, scrollToIndex, heights);
+    }
+
+    public static void verifyListViewScrollTo(ListView listView, int listViewHeight, int scrollToIndex, Integer[] heights) {
+        double sumOfHeights = 0;
+        double viewportLength = listViewHeight - 2; // it would be better to calculate this from listView but there is no API for this
+
+        for(int height: heights) {
+            sumOfHeights += height;
+        }
+        double sizeBelow = 0;
+        for(int i = scrollToIndex; i < heights.length; i += 1) {
+            sizeBelow += heights[i];
+        }
+
+        IndexedCell<Integer> firstCell = VirtualFlowTestUtils.getCell(listView, 0);
+        IndexedCell<Integer> scrollToCell = VirtualFlowTestUtils.getCell(listView, scrollToIndex);
+        IndexedCell<Integer> lastCell = VirtualFlowTestUtils.getCell(listView, heights.length - 1);
+
+        double lastCellSize = heights[heights.length -1];
+
+        boolean scrolledToTop = scrollToIndex == 0;
+        boolean scrolledToBottomElement = scrollToIndex == heights.length - 1;
+        boolean scrolledToBottom = false;
+        boolean isLastIndex = scrollToIndex == heights.length - 1;
+        boolean shouldScrollToBottom = (sizeBelow < viewportLength) || (isLastIndex && lastCellSize >= viewportLength);
+
+        if (Math.abs(lastCell.getLayoutY() - (viewportLength - lastCellSize)) <= 1.0) {
+            scrolledToBottom = true;
+        }
+
+        assertTrue("Our cell must be visible!", scrollToCell.isVisible());
+
+        if(lastCell.isVisible() && sumOfHeights >= viewportLength) {
+            // There shouldn't be any space between the last cell and the bottom
+            assertTrue("Last cell shouldn't leave space between itself and the bottom", lastCell.getLayoutY() + 1 > (viewportLength - lastCellSize));
+        }
+        if(sumOfHeights < viewportLength) {
+            // If we have less cells then space, then all cells are shown, and the position of the last cell, is the sum of the height of the previous cells.
+            assertEquals("Last cell should be at the bottom, if we scroll to it", lastCell.getLayoutY(), sumOfHeights - lastCellSize,1.);
+        }
+        if(shouldScrollToBottom && sumOfHeights > viewportLength) {
+            // If we scroll to the bottom, then the last cell should be exactly at the bottom
+            // assertEquals("", lastCell.getLayoutY(), viewportLength - lastCellSize,1.);
```

Comment:
Expected and actual args are backwards.

Reply:
this particular assert was commented out, and I removed it. I reverted the args in the previous one though.